### ISSUE TITLE
Confusion over the meaning of 'rotating about an axis' in PyTorch3d

### ISSUE THREAD

**Issue by DanielTakeshi** [facebookresearch/pytorch3d#1192]
## ❓ Questions on how to use PyTorch3D

I want to make sure I fully understand the coordinate system in PyTorch3D. I am trying to rotate points about an axis. But the points do not seem to be transformed the way I expect them to (they rotate in the opposite direction). Here is a MWE:

```
import torch
import numpy as np
from pytorch3d.transforms import (Rotate, axis_angle_to_matrix,)
np.set_printoptions(suppress=True, precision=4, linewidth=150, edgeitems=20)
torch.set_printoptions(sci_mode=False, precision=4, linewidth=150, edgeitems=20)
DEG_TO_RAD = np.pi / 180


if __name__ == "__main__":
    # Hard-coded data.
    xyz_np = np.array([[
        [0, 0, 0],
        [0, 1, 0],
        [0, 2, 0],
        [0, 3, 0],
        [0, 4, 0],
        [0, 5, 0],
        [0, 6, 0],
        [0, 7, 0],
        [0, 8, 0],
        [0, 9, 0],
    ]])
    xyz = torch.as_tensor(xyz_np).float()

    # Define a transformation.
    alpha = 45. * DEG_TO_RAD
    trans = torch.as_tensor(np.array([[0., 0., 0.]]))
    axis_angle = torch.as_tensor(np.array([[0., 0., alpha]]))
    Rot = axis_angle_to_matrix(axis_angle)
    T_transform = Rotate(Rot).translate(trans)
    print(f'Applying this transform (also alpha={alpha:0.3f} rad.):')
    print(f'{Rot}\n{trans}\n')

    # Apply the transform on the set of points.
    # Shapes of (xyz,xyz_next): torch.Size([1, 5, 3]) torch.Size([1, 5, 3])
    xyz_next = T_transform.transform_points(xyz)

    # In a RH coordinate system, shouldn't the x values be negative
    # if we are rotating about the positive z axis?
    print(f'xyz:\n{xyz}')
    print(f'xyz transformed:\n{xyz_next}')
```

TL;DR I want a 45 degree rotation about the z axis. PyTorch3D docs are here:
https://pytorch3d.readthedocs.io/en/latest/modules/transforms.html
And whenever it refers to axis angle, it says thef rotation magnitude is anticlockwise in the direction of the axis.
![Screenshot from 2022-05-07 21-13-08](https://user-images.githubusercontent.com/5532096/167277627-8b091e17-466c-4740-b1fe-a308ec41f0fe.png)

Here is the output:

```
$ python simple_pytorch3d_test_github.py 
Applying this transform (also alpha=0.785 rad.):
tensor([[[ 0.7071, -0.7071,  0.0000],
         [ 0.7071,  0.7071,  0.0000],
         [ 0.0000,  0.0000,  1.0000]]], dtype=torch.float64)
tensor([[0., 0., 0.]], dtype=torch.float64)

xyz:
tensor([[[0., 0., 0.],
         [0., 1., 0.],
         [0., 2., 0.],
         [0., 3., 0.],
         [0., 4., 0.],
         [0., 5., 0.],
         [0., 6., 0.],
         [0., 7., 0.],
         [0., 8., 0.],
         [0., 9., 0.]]])
xyz transformed:
tensor([[[0.0000, 0.0000, 0.0000],
         [0.7071, 0.7071, 0.0000],
         [1.4142, 1.4142, 0.0000],
         [2.1213, 2.1213, 0.0000],
         [2.8284, 2.8284, 0.0000],
         [3.5355, 3.5355, 0.0000],
         [4.2426, 4.2426, 0.0000],
         [4.9497, 4.9497, 0.0000],
         [5.6569, 5.6569, 0.0000],
         [6.3640, 6.3640, 0.0000]]])
(svd_test) seita@starship:~/flow_SVD_MWE (master) $ 
```

In a right-handed coordinate system, where  x=thumb, y=index, middle=z, shouldn't the resulting transformed points have negative x values? The input points are all pointing in the direction of the y-axis, then when we rotate in a positive direction (anticlockwise about z) wouldn't that mean all the points have negative x coordinate?  Here is a drawing that visualizes what I thought would happen (left) and what actually happened (right):
![Screenshot from 2022-05-07 21-27-42](https://user-images.githubusercontent.com/5532096/167277972-3cef8ab1-0c64-4a0d-98c2-dd7bce1e2b6f.png)

There is a related issue here that I followed up on:
https://github.com/facebookresearch/pytorch3d/issues/625

My main question is that I want to make sure I understand what "rotation about an axis" and "positive rotation" means in PyTorch3D. I had always assumed that, if my thumb is pointing in the direction of the positive axis, that the way my fingers curl is the direction of positive rotation (as shown here for example https://en.wikipedia.org/wiki/Right-hand_rule#Rotations http://faculty.salina.k-state.edu/tim/robotics_sg/Pose/Trans3d.html and other places). Is that not the case here (or am I misunderstanding something)?

**Comment by DanielTakeshi** [comment]
Following up on this, here is an example using pyquaternion http://kieranwynn.github.io/pyquaternion/  where I use a rotation with the same initialization (axis-angle in the positive z direction):

```
from pyquaternion import Quaternion
import numpy as np
np.set_printoptions(suppress=True, precision=4, linewidth=150, edgeitems=20)
DEG_TO_RAD = np.pi / 180


if __name__ == "__main__":
    # Hard-coded data.
    xyz = np.array([0, 1, 0])

    # Define a transformation.
    alpha = 45. * DEG_TO_RAD
    my_quaternion = Quaternion(axis=[0, 0, 1], angle=alpha)

    # Apply it.
    xyz_next = my_quaternion.rotate(xyz)
    print(f'xyz:\n{xyz}')
    print(f'xyz transformed:\n{xyz_next}')
```

Here the print output is:

```
$ python simple_pyquaternion_test_github.py 
xyz:
[0 1 0]
xyz transformed:
[-0.7071  0.7071  0.    ]
```

This is the behavior I was expecting to see: a positive rotation about the z-axis transforms it so that (0,1,0) has a negative x coordinate.

**Comment by bottler** [comment]
The convention for rotation matrices in rotation_conversions is that they apply to column matrices (by premultiplication). The convention for all matrices in Transform3D is that they apply to row matrices (by postmultiplication). Unfortunately `Rotate(Rot)` should be `Rotate(Rot.T)`. We could add a mention of this to the documentation of Rotate.

**Comment by DanielTakeshi** [comment]
Thanks!
